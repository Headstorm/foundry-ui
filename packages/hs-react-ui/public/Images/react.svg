<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 19.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_2" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 width="561.8px" height="499.9px" viewBox="-80.4 151.2 561.8 499.9" style="enable-background:new -80.4 151.2 561.8 499.9;"
	 xml:space="preserve">
<style type="text/css">
	.st0{fill:#939598;}
</style>
<circle class="st0" cx="200.5" cy="400.6" r="50.2"/>
<path class="st0" d="M200.5,515.6c-70.5,0-132.1-8.3-178.2-24.1c-29.9-10.2-55.3-23.8-73.4-39.3c-19.2-16.4-29.4-34.3-29.4-51.6
	c0-33.2,36.4-65.7,97.5-86.9c50-17.4,115.2-27.1,183.4-27.1c67,0,131.3,9.4,181,26.6c29.1,10,53.6,23,71,37.4
	c18.9,15.8,28.9,33.1,28.9,50c0,34.5-40.7,69.4-106.3,91.1C328.6,507.1,266.6,515.6,200.5,515.6z M200.5,310.6
	c-64.7,0-128.7,9.4-175.5,25.7c-40.5,14.1-81.4,39.5-81.4,64.3c0,10,7.4,21.8,21,33.4c15.8,13.5,38.4,25.6,65.5,34.8
	c43.6,14.9,102.6,22.8,170.4,22.8c63.6,0,122.9-8,167-22.7c61.7-20.5,89.9-49.8,89.9-68.3c0-9.5-7.2-20.7-20.3-31.6
	c-15.1-12.6-37.1-24.1-63.4-33.2C326.4,319.6,264.9,310.6,200.5,310.6z"/>
<path class="st0" d="M86.6,650.8c-10.2,0-19.2-2.2-26.8-6.6C31,627.6,21.1,579.8,33.1,516.3c9.9-52.1,34.1-113.3,68.2-172.4h0
	c33.5-58,73.7-109,113.4-143.5c23.2-20.2,46.7-35,67.9-42.8c23.1-8.5,43.1-8.5,57.7-0.1c29.9,17.2,39.8,70,25.8,137.6
	c-9.9,48-33.5,105.9-66.5,163.2c-35.2,61-73.2,110.2-109.9,142.3c-23.8,20.8-48.3,36-70.7,43.9C107.4,648.7,96.4,650.8,86.6,650.8z
	 M122.1,355.9c-32.3,56-56.2,116.1-65.4,164.9c-8,42.1-6.5,90.2,15,102.6c16.1,9.3,55,0.5,102.3-40.9c34.7-30.3,71-77.5,104.9-136.2
	c31.8-55.1,54.4-110.5,63.8-156c6-29.1,7.3-55.5,3.8-76.4c-3-17.9-9.4-30.6-18.2-35.6c-8.2-4.7-21.5-4.1-37.5,1.8
	c-18.5,6.8-39.4,20.1-60.4,38.4C192.8,251.3,154.3,300.1,122.1,355.9L122.1,355.9z"/>
<path class="st0" d="M314.4,651.1c-27.2,0-61.7-16.4-97.7-47.4c-40.2-34.6-81.1-86.1-115.3-145.2v0c-33.6-58-57.6-118.3-67.7-170
	c-5.9-30.2-7-57.9-3.2-80.2c4.2-24.3,14.1-41.6,28.8-50.1c29.8-17.3,80.5,0.5,132.1,46.4c36.6,32.5,75,81.9,108.1,139.1
	c35.3,61,59,118.5,68.4,166.3c6.1,31,7.1,59.8,2.8,83.2c-4.6,24.9-15,42.6-30,51.3C333.2,648.9,324.3,651.1,314.4,651.1z
	 M122.2,446.5c32.4,56,72.6,106.7,110.2,139c32.5,28,74.9,50.6,96.4,38.2c16.1-9.3,27.9-47.4,15.7-109
	c-9-45.2-31.7-100.2-65.6-158.9c-31.9-55.1-68.6-102.4-103.3-133.2c-22.2-19.7-44.4-34.1-64.2-41.5c-17-6.4-31.2-7.1-39.9-2.1
	c-8.2,4.7-14.3,16.6-17.2,33.4c-3.3,19.4-2.3,44.2,3.1,71.5C66.9,333,89.9,390.7,122.2,446.5L122.2,446.5z"/>
</svg>
