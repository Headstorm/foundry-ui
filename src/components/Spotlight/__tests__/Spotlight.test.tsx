import React from 'react';
import { render, waitFor } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import { Button, Spotlight } from '../../../index';

expect.extend(toHaveNoViolations);

describe('Spotlight', () => {
  it('Should pass accessibility test with default props', async () => {
    const component = <Spotlight />;
    const { container } = render(component);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
  it('Should match previous snapshot with default props', async () => {
    const onEnd = jest.fn();
    const component = <Spotlight onAnimationEnd={onEnd} />;
    const { container } = render(component, { container: document.body });
    await waitFor(() => expect(onEnd).toHaveBeenCalled());
    expect(container).toMatchSnapshot();
  });
  it('Should match previous snapshot with a targetElement', async () => {
    const buttonRef = React.createRef<HTMLButtonElement>();
    const onEnd = jest.fn();
    const component = (
      <Spotlight onAnimationEnd={onEnd} targetElement={buttonRef.current}>
        <Button containerRef={buttonRef}>Test</Button>
      </Spotlight>
    );
    const { container } = render(component, { container: document.body });
    await waitFor(() => buttonRef.current);
    await waitFor(() => expect(onEnd).toHaveBeenCalled());
    expect(container).toMatchSnapshot();
  });
});
