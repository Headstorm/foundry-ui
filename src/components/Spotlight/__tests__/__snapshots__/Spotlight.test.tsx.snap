// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spotlight Should match previous snapshot with a targetElement 1`] = `
<body>
  .c9 {
  opacity: 0;
  background: linear-gradient( 90deg, rgba(0,0,0,0.75), rgba(0,0,0,0.25), rgba(0,0,0,0.75) ) repeat;
  color: transparent;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-size: 100vw 100vh;
  background-attachment: fixed;
  border-radius: 0.25rem;
  -webkit-animation: gZwmnl 2s linear infinite;
  animation: gZwmnl 2s linear infinite;
}

.c3 {
  display: block;
  position: relative;
}

.c3 > * {
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
}

.c5 {
  position: relative;
}

.c7 {
  pointer-events: none;
  position: absolute;
  top: 0;
  left: 0;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  font-size: 1em;
  padding: .75em 1em;
  border-radius: 0.25em;
  -webkit-transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  transition: background-color .2s, color .5s, outline .5s, filter .5s, box-shadow .5s;
  contain: layout;
  outline: 0 none;
  border: 1px solid transparent;
  cursor: pointer;
  background-color: #9FA2A5;
  color: #000;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4:hover {
  background-color: #929599;
}

.c4:focus {
  outline: none;
  box-shadow: 0 0 5px 0.150rem #338BAA;
}

.c6 {
  position: absolute;
  width: 100%;
  height: 100%;
  left: 0;
  top: 0;
}

.c8 {
  border-radius: 0.25em;
}

.c0 {
  height: 100%;
  width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  contain: strict;
  -webkit-clip-path: url(#foundryMask);
  clip-path: url(#foundryMask);
}

.c1 {
  position: fixed;
  z-index: -1;
}

.c2 {
  display: inline-block;
  position: fixed;
  top: 0;
  left: 0;
  padding-bottom: 1rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  max-width: 50vw;
}

<div>
    <div
      class=" c0"
      globalstyles="
  
"
      style="background-color: rgba(0, 0, 0, 0.7);"
    >
      <div
        class=" c1"
        globalstyles="
  
"
        style="top: 0px; left: 0px; width: 1024px; height: 383px;"
      />
      <div
        class=" c1"
        globalstyles="
  
"
        style="top: 384px; left: 0px; width: 1024px; height: 383px;"
      />
      <div
        class=" c1"
        globalstyles="
  
"
        style="top: 384px; left: 0px; width: 512px; height: 2px;"
      />
      <div
        class=" c1"
        globalstyles="
  
"
        style="top: 384px; right: 0px; width: 512px; height: 2px;"
      />
    </div>
    <svg
      height="0"
      style="position: fixed; top: 0px; left: 0px;"
      viewBox="0 0 1024 768"
      width="0"
    >
      <defs>
        <clippath
          clip-rule="evenodd"
          id="foundryMask"
        >
          <path
            d="M 0 0 h1024 v768 h-1024 
    M 512 384
    A 0, 0, 0, 1, 1, 512 385
    L 512 384
  "
            fill="#FFFFFF"
          />
        </clippath>
      </defs>
    </svg>
    <div
      class=" c2"
      globalstyles="
  
"
      style="transform: translate(512px, 384px) translate(0%, -100%);"
    >
      <button
        class=" c3 c4"
        color="#9FA2A5"
        elevation="0"
        role="button"
        type="button"
      >
        Test
        <div
          class="c5 c6"
        >
          <svg
            class="c7 c8"
            height="0px"
            viewBox="0 0 0 0"
            width="0px"
          />
        </div>
        <div
          class=" c9"
          color="#000"
        />
      </button>
    </div>
  </div>
</body>
`;

exports[`Spotlight Should match previous snapshot with default props 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  contain: strict;
  -webkit-clip-path: url(#foundryMask);
  clip-path: url(#foundryMask);
}

.c1 {
  position: fixed;
  z-index: -1;
}

.c2 {
  display: inline-block;
  position: fixed;
  top: 0;
  left: 0;
  padding-bottom: 1rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  max-width: 50vw;
}

<body>
  <div>
    <div
      class=" c0"
      globalstyles="
  
"
      style="background-color: rgba(0, 0, 0, 0.7);"
    >
      <div
        class=" c1"
        globalstyles="
  
"
        style="top: 0px; left: 0px; width: 1024px; height: 383px;"
      />
      <div
        class=" c1"
        globalstyles="
  
"
        style="top: 384px; left: 0px; width: 1024px; height: 383px;"
      />
      <div
        class=" c1"
        globalstyles="
  
"
        style="top: 384px; left: 0px; width: 512px; height: 2px;"
      />
      <div
        class=" c1"
        globalstyles="
  
"
        style="top: 384px; right: 0px; width: 512px; height: 2px;"
      />
    </div>
    <svg
      height="0"
      style="position: fixed; top: 0px; left: 0px;"
      viewBox="0 0 1024 768"
      width="0"
    >
      <defs>
        <clippath
          clip-rule="evenodd"
          id="foundryMask"
        >
          <path
            d="M 0 0 h1024 v768 h-1024 
    M 512 384
    A 0, 0, 0, 1, 1, 512 385
    L 512 384
  "
            fill="#FFFFFF"
          />
        </clippath>
      </defs>
    </svg>
    <div
      class=" c2"
      globalstyles="
  
"
      style="transform: translate(512px, 384px) translate(0%, -100%);"
    />
  </div>
</body>
`;
